R/utilities-helper.R

```diff
@@ -0,0 +1,69 @@
+#' Enumerate With Values
+#'
+#' Creates an enumeration from specified values, facilitating value validation.
+#'
+#' @param enumValues A named list where each name corresponds to a parameter and its allowed values.
+#'
+#' @return A list where each element is named by the parameter and contains its allowed values.
+#' @examples
+#' options <- list(
+#'   residualWeightingMethod = c("none", "std", "mean", "error"),
+#'   robustMethod = c("none", "huber", "bisquare")
+#' )
+#' myOptions <- enumWithValues(options)
+#' @export
+enumWithValues <- function(enumValues) {
+  myEnum <- list()
+  for (param in names(enumValues)) {
+    allowedValues <- enumValues[[param]]
+    myEnum[[param]] <- allowedValues
+  }
+  return(myEnum)
+}
+
+#' Validate Input Options
+#'
+#' Validates the specified input options against a set of valid options and their allowed values or types.
+#'
+#' @param inputOptions A list of input options to validate.
+#' @param validOptions A list of valid options with their allowed values or types.
+#'
+#' @return None; this function stops execution with an error message if validation fails.
+#' @examples
+#' validOptions <- list(
+#'   option1 = c("value1", "value2"),
+#'   option2 = "boolean",
+#'   option3 = "numeric"
+#' )
+#' inputOptions <- list(option1 = "value1", option2 = TRUE, option3 = 5)
+#' validateOptions(inputOptions, validOptions)
+#' @export
+validateIsOption <- function(inputOptions, validOptions) {
```

Comment:
I think this is a very useful function and belongs to the ospsuite.rutils package. Can you implement it here: https://github.com/Open-Systems-Pharmacology/OSPSuite.RUtils ?